src/autopas/selectors/tuningStrategy/PredictiveTuning.h

```diff
@@ -0,0 +1,370 @@
+
+  /**
+   * Stores the traversal times for each configuration.
+   */
+  std::unordered_map<Configuration, std::vector<std::pair<int, size_t>>, ConfigHash> _traversalTimesStorage;
+
+  /**
+   * Contains the predicted time for each configuration.
+   */
+  std::unordered_map<Configuration, size_t, ConfigHash> _traversalPredictions;
+
+  /**
+   * Contains the configuration that are going to be tested.
+   */
+  std::set<Configuration> _optimalSearchSpace;
+
+  /**
+   * Contains the configurations that are not invalid the whole optimalSearchSpace was invalid
+   */
+  std::set<Configuration> _validSearchSpace;
+
+  /**
+   * Gets incremented after every completed tuning phase.
+   * Mainly used for the traversal time storage.
+   */
+  int _timer = 0;
+
+  /**
+   * Checks if a valid configuration was in the optimal SearchSpace.
+   */
+  bool _validConfiguration = false;
+};
+
+void PredictiveTuning::populateSearchSpace(const std::set<ContainerOption> &allowedContainerOptions,
+                                           const std::set<double> &allowedCellSizeFactors,
+                                           const std::set<TraversalOption> &allowedTraversalOptions,
+                                           const std::set<DataLayoutOption> &allowedDataLayoutOptions,
+                                           const std::set<Newton3Option> &allowedNewton3Options) {
+  // generate all potential configs
+  for (auto &containerOption : allowedContainerOptions) {
+    // get all traversals of the container and restrict them to the allowed ones
+    const std::set<TraversalOption> &allContainerTraversals =
+        compatibleTraversals::allCompatibleTraversals(containerOption);
+    std::set<TraversalOption> allowedAndApplicable;
+    std::set_intersection(allowedTraversalOptions.begin(), allowedTraversalOptions.end(),
+                          allContainerTraversals.begin(), allContainerTraversals.end(),
+                          std::inserter(allowedAndApplicable, allowedAndApplicable.begin()));
+
+    for (auto &cellSizeFactor : allowedCellSizeFactors)
+      for (auto &traversalOption : allowedAndApplicable) {
+        for (auto &dataLayoutOption : allowedDataLayoutOptions) {
+          for (auto &newton3Option : allowedNewton3Options) {
+            _searchSpace.emplace(containerOption, cellSizeFactor, traversalOption, dataLayoutOption, newton3Option);
+          }
+        }
+      }
+  }
+
+  AutoPasLog(debug, "Points in search space: {}", _searchSpace.size());
+
+  if (_searchSpace.empty()) {
+    autopas::utils::ExceptionHandler::exception("PredictiveTuning: No valid configurations could be created.");
+  }
+
+  for (auto &configuration : _searchSpace) {
+    std::vector<std::pair<int, size_t>> vector;
+    _traversalTimesStorage.emplace(configuration, vector);
+  }
+
+  _currentConfig = _searchSpace.begin();
+}
+
+void PredictiveTuning::selectPredictedConfigurations() {
+  if (_searchSpace.size() == 1 || _timer == 0 || _timer == 1) {
```

Comment:
```suggestion
  if (_searchSpace.size() == 1 || _timer < 2) {
```
So if I understand this correctly, this condition is responsible for this strategy only shrinking the search space starting from the third tuning phase?
If so, make the `2` a variable.
Else explain what I missed ;)